pkg/rpcclient/rpc.go

```diff
@@ -1285,3 +1285,44 @@ func (c *Client) TerminateSession(sessionID uuid.UUID) (bool, error) {
 
 	return resp, nil
 }
+
+// GetRawNotaryTransaction  returns main or fallback transaction from the
+// RPC node's notary request pool. This function uses a `[]bytes` result
+// from `getrawnotarytransaction`.
+func (c *Client) GetRawNotaryTransaction(hash util.Uint256) (*transaction.Transaction, error) {
+	var (
+		params = []any{hash.StringLE()}
+		resp   []byte
+		err    error
+	)
+	if err = c.performRequest("getrawnotarytransaction", params, resp); err != nil {
+		return nil, err
+	}
+	return transaction.NewTransactionFromBytes(resp)
+}
+
+// GetRawNotaryTransactionVerbose returns main or fallback transaction from the
+// RPC node's notary request pool. This function uses verbose
+// (*transaction.Transaction) result from `getrawnotarytransaction`.
+func (c *Client) GetRawNotaryTransactionVerbose(hash util.Uint256) (*transaction.Transaction, error) {
+	var (
+		params = []any{hash.StringLE(), 1} // 1 for verbose.
+		resp   = &transaction.Transaction{}
+		err    error
+	)
+	if err = c.performRequest("getrawnotarytransaction", params, resp); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// GetRawNotaryPool returns hashes of main P2PNotaryRequest transactions that
+// are currently in the RPC node's notary request pool with the corresponding
+// hashes of fallback transactions.
+func (c *Client) GetRawNotaryPool() (*result.RawNotaryPool, error) {
```

Comment:
Can we add tests for these Client APIs to `pkg/rpcclient/rpc_test.go`?

And we also need an end-to-end test to be added to `pkg/services/rpcsrv/client_test.go`. Check how `TestNotaryActor` or `TestSignAndPushP2PNotaryRequest` are implemented there. Do not use the old `SignAndPushP2PNotaryRequest` Client API for this test, use notary Actor instead.

Reply:
We probably can, so I asked in https://github.com/nspcc-dev/neo-go/pull/3098#discussion_r1306712301 I found out that SubmitP2PNotaryRequest wasn't test covered. So it's a little unclear what we want to test.

Reply:
What we need within the scope of this issue is to create test for these new client's API. Leave the `SubmitP2PNotaryRequest` for future issues.

See the https://github.com/nspcc-dev/neo-go/pull/3099/commits/617c628c242b3f1778c42c60f96a33344daf45d7, that's exactly you have to do with these new clients APIs.

Reply:
> Can we add tests for these Client APIs to pkg/rpcclient/rpc_test.go?

Done, check, please. Is it ok?

>And we also need an end-to-end test to be added to pkg/services/rpcsrv/client_test.go. Check how TestNotaryActor or TestSignAndPushP2PNotaryRequest are implemented there. Do not use the old SignAndPushP2PNotaryRequest Client API for this test, use notary Actor instead.

Haven't done yet.

Reply:
> Done, check, please. Is it ok?

It needs to be adjusted a bit, I've commented there. Otherwise OK.

Reply:
Fixed tests in pkg/rpcclient/rpc_test.go and added tests in pkg/services/rpcsrv/client_test.go.